.github/workflows/python_package.yml

```diff
@@ -76,10 +76,36 @@ jobs:
           export PATH=${CONDA}/bin:${PATH}
           $GITHUB_WORKSPACE/.ci/setup.sh || exit -1
           $GITHUB_WORKSPACE/.ci/test.sh || exit -1
+  test-oldest-versions:
+    name: Python - oldest supported versions (ubuntu-latest)
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: Create wheel
+        run: |
+          docker run \
+            --rm \
+            -v $(pwd):/opt/lgb-build \
+            -w /opt/lgb-build \
+            lightgbm/vsts-agent:manylinux_2_28_x86_64 \
```

Comment:
This is the same image we use for building the wheels published to PyPI.

https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/.vsts-ci.yml#L21-L22

https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/.vsts-ci.yml#L36-L38

It's also not possible, as of #5759, to run `sh build-python.sh` in `Python < 3.7`, since:

* `scikit-build-core` has a floor of `Python >= 3.7`
* the oldest supported `setuptools` for Python 3.6, `v59.6.0` ([December 2021](https://pypi.org/project/setuptools/#history)), doesn't support `pyproject.toml`

So as of #5759, Python 3.6 users will have to use the published wheels. Customizations like `--cuda` will require building the wheel in a Python 3.7+ environment and then distributing it into wherever that Python 3.6 code is running.

I think Python 3.6 users should be expect to face issues for many packages (not just LightGBM) as a result of the changes in the Python packaging ecosystem described in #5061.